dataflow/src/main/java/org/checkerframework/dataflow/expression/JavaExpression.java

```diff
@@ -660,4 +677,159 @@ public static JavaExpression getPseudoReceiver(TreePath path, TypeMirror enclosi
+        List<JavaExpression> argumentsJe =
+                argumentTreesToJavaExpressions(newClassTree, newClassTree.getArguments());
+
+        JavaExpression receiverJe = JavaExpression.getReceiver(newClassTree);
+        return ViewpointAdaptJavaExpression.viewpointAdapt(this, receiverJe, argumentsJe);
+    }
+
+    /**
+     * Converts method or constructor arguments from Trees to JavaExpressions, accounting for
+     * varargs.
+     *
+     * @param tree invocation of the method or constructor
+     * @param argTrees the arguments to the method or constructor; subexpressions of {@code tree}
+     * @return the arguments, as JavaExpressions
+     */
+    private static List<JavaExpression> argumentTreesToJavaExpressions(
+            ExpressionTree tree, List<? extends ExpressionTree> argTrees) {
+
+        if (tree.getKind() == Kind.METHOD_INVOCATION) {
+            MethodInvocationTree methodInvoc = (MethodInvocationTree) tree;
+            assert TreeUtils.isUseOfElement(methodInvoc) : "@AssumeAssertion(nullness): tree kind";
+            ExecutableElement method = TreeUtils.elementFromUse(methodInvoc);
+            if (isVarArgsInvocation(method, argTrees)) {
+                List<JavaExpression> result = new ArrayList<>(method.getParameters().size());
+
+                for (int i = 0; i < method.getParameters().size() - 1; i++) {
+                    result.add(JavaExpression.fromTree(argTrees.get(i)));
+                }
+                List<JavaExpression> varargArgs =
+                        new ArrayList<>(argTrees.size() - method.getParameters().size() + 1);
+                for (int i = method.getParameters().size() - 1; i < argTrees.size(); i++) {
+                    varargArgs.add(JavaExpression.fromTree(argTrees.get(i)));
+                }
+                Element varargsElement =
+                        method.getParameters().get(method.getParameters().size() - 1);
+                TypeMirror tm = ElementUtils.getType(varargsElement);
+                result.add(new ArrayCreation(tm, Collections.emptyList(), varargArgs));
+
+                return result;
+            }
+        }
+
+        List<JavaExpression> result = new ArrayList<>(argTrees.size());
+        for (ExpressionTree argTree : argTrees) {
+            result.add(JavaExpression.fromTree(argTree));
+        }
+        return result;
+    }
+
+    /**
+     * Returns true if method is a varargs method or constructor and its varargs arguments are not
+     * passed in an array.
+     *
+     * @param method the method or constructor
+     * @param args the arguments at the call site
+     * @return true if method is a varargs method and its varargs arguments are not passed in an
+     *     array
+     */
+    private static boolean isVarArgsInvocation(
+            ExecutableElement method, List<? extends ExpressionTree> args) {
+        if (method == null || !method.isVarArgs()) {
```

Comment:
How can `method` be null?  If it is, I think there should be an exception thrown here or in the caller.